OAuthLab/app/src/main/java/com/scottlindley/oauthlab/MainActivity.java

```diff
@@ -0,0 +1,155 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
+        final NetworkInfo info = connectivityManager.getActiveNetworkInfo();
+
+        if (info != null && info.isConnected()) {
+            String credentials = CONSUMER_KEY + ":" + CONSUMER_SECRET;
+            mEncodedCredentials = new String(Base64.encodeBase64(credentials.getBytes()));
+//                    Base64.encodeBase64String(credentials.getBytes());
+            obtainBearerToken();
+        } else {
+            Toast.makeText(this, "No network connection", Toast.LENGTH_SHORT).show();
+        }
+        mTweets = new ArrayList<>();
+        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(MainActivity.this));
+        mRecyclerView.setAdapter(new RecyclerAdapter(mTweets));
+
+        mEditQuery = (EditText) findViewById(R.id.edit_query);
+        mSearchButton = (Button) findViewById(R.id.search_button);
+
+        mSearchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (info != null && info.isConnected()) {
+                    performUserSearch(mEditQuery.getText().toString());
+                }
+            }
+        });
+
+    }
+
+    private void obtainBearerToken() {
+        OkHttpClient client = new OkHttpClient();
+        Headers headers = new Headers.Builder()
+                .add("Authorization", "Basic " + mEncodedCredentials)
+                .add("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+                .build();
+        RequestBody body = new FormBody.Builder()
+                .add("grant_type", "client_credentials")
+                .build();
+        Request request = new Request.Builder()
+                .url("https://api.twitter.com/oauth2/token")
+                .headers(headers)
+                .post(body)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                try {
+                    JSONObject responseObject = new JSONObject(response.body().string());
+                    mBearerToken = responseObject.getString("access_token");
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    private void performUserSearch(String query){
+        query = "https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name="+query;
```

Comment:
Might be helpful to put this URL in a constant at the top of the class, just to make it easier to find